components/indiefunnels/sources/common/base.mjs

```diff
@@ -0,0 +1,34 @@
+import crypto from "crypto";
+import indiefunnels from "../../indiefunnels.app.mjs";
+
+export default {
+  props: {
+    indiefunnels,
+    db: "$.service.db",
+    http: "$.interface.http",
+  },
+  hooks: {
+    async activate() {
+      const response = await this.indiefunnels.createHook({
+        data: {
+          target: this.http.endpoint,
+          secret: crypto.randomUUID(),
+          events: this.getEvent(),
+        },
+      });
+      this.db.set("webhookId", response.id);
+    },
+    async deactivate() {
+      const webhookId = this.db.get("webhookId");
```

Comment:
Need to use getter/setter methods for interacting with `db`. E.g. `_getWebhookId()`, `_setWebhookId()`.